presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -474,7 +474,14 @@ public StreamProperties visitTopN(TopNNode node, List<StreamProperties> inputPro
         @Override
         public StreamProperties visitSort(SortNode node, List<StreamProperties> inputProperties)
         {
-            return StreamProperties.ordered();
+            StreamProperties sourceProperties = Iterables.getOnlyElement(inputProperties);
+            if (sourceProperties.isSingleStream()) {
+                // stream is only sorted if sort operator is executed without parallelism
+                return StreamProperties.ordered();
```

Comment:
Hm... btw, do you know why the return `StreamProperties` do not encode on what column(s) the data is sorted?